rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -145,7 +145,9 @@ public synchronized void addNode(Node node) {
                 // adding one node from each host
                 for (InetAddress h : nodesOnHost.keySet()) {
                     // always have at least one node on each host
-                    toPing.add(nodesOnHost.get(h).iterator().next());
+                    if (nodesOnHost.get(h) != null && nodesOnHost.get(h).size() > 0) {
+                        toPing.add(nodesOnHost.get(h).iterator().next());
+                    }
                 }
             }
             HashMap<InetAddress, Long> hostsTopology = null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The type of the "hostsTopology" object should be an interface such as "Map" rather than the implementation "HashMap". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1319)